### DISCUSSION TITLE
"Granian Asgi [POST] (c256)" and file responses "Granian (pathsend) (c128)" are slower than with Uvicorn.

### DISCUSSION THREAD

**Discussion by itsmeadarsh2008** [emmett-framework/granian#289] [General]
Please someone investigate this. I don't know Rust
I'll be grateful if you could fix this
You can convert it to a discussion if you want to
See: [benchmarks/vs.md](https://github.com/emmett-framework/granian/blob/master/benchmarks/vs.md)

**Comment by gi0baro** [comment]
I'm not sure why you state pathsend is slower than uvicorn when it's actually faster..

Anyways, there's nothing really to "fix", this is simply the current state of Granian; the project aim was never set to be the fastest server out there.
Improvements might still be made in performance, but it's not the main priority right now.

**Comment by itsmeadarsh2008** [comment]
I was talking about the max-latency being higher in granian compared to uvicorn.

**Reply by gi0baro** [reply]
Both the average latency and RPS are better though 😅

**Reply by itsmeadarsh2008** [reply]
I'm here after this discussion after the proposal of replacing uvicorn of nicegui with your project.
https://github.com/zauberzeug/nicegui/discussions/3039#discussioncomment-9367211
